pkg/riff/commands/requestprocessor_invoke_test.go

```diff
@@ -0,0 +1,408 @@
+					"curl localhost -H 'Host: test-requestprocessor.example.com' -w '\n'\n",
+				} {
+					if !strings.Contains(output, expected) {
+						t.Errorf("expected command output to contain %q, actually %q", expected, output)
+					}
+				}
+			},
+		},
+		{
+			Name: "unknown ingress",
+			Args: []string{requestprocessorName},
+			GivenObjects: []runtime.Object{
+				requestProcessor,
+				&corev1.Service{
+					ObjectMeta: metav1.ObjectMeta{
+						Namespace: "istio-system",
+						Name:      "istio-ingressgateway",
+					},
+					Spec: corev1.ServiceSpec{
+						Ports: []corev1.ServicePort{},
+					},
+				},
+			},
+			ShouldError: true,
+		},
+		{
+			Name:       "missing ingress",
+			Args:       []string{requestprocessorName},
+			ExecHelper: "RequestProcessorInvoke",
+			GivenObjects: []runtime.Object{
+				requestProcessor,
+			},
+			ShouldError: true,
+		},
+		{
+			Name: "request processor not ready",
+			Args: []string{requestprocessorName},
+			GivenObjects: []runtime.Object{
+				&requestv1alpha1.RequestProcessor{
+					ObjectMeta: metav1.ObjectMeta{
+						Namespace: defaultNamespace,
+						Name:      requestprocessorName,
+					},
+					Status: requestv1alpha1.RequestProcessorStatus{
+						Domain: fmt.Sprintf("%s.example.com", requestprocessorName),
+					},
+				},
+				ingressService,
+			},
+			ShouldError: true,
+		},
+		{
+			Name: "request processor missing domain",
+			Args: []string{requestprocessorName},
+			GivenObjects: []runtime.Object{
+				&requestv1alpha1.RequestProcessor{
+					ObjectMeta: metav1.ObjectMeta{
+						Namespace: defaultNamespace,
+						Name:      requestprocessorName,
+					},
+					Status: requestv1alpha1.RequestProcessorStatus{
+						Status: duckv1alpha1.Status{
+							Conditions: []duckv1alpha1.Condition{
+								{Type: requestv1alpha1.RequestProcessorConditionReady, Status: "True"},
+							},
+						},
+					},
+				},
+				ingressService,
+			},
+			ShouldError: true,
+		},
+		{
+			Name: "missing request processor",
+			Args: []string{requestprocessorName},
+			GivenObjects: []runtime.Object{
+				ingressService,
+			},
+			ShouldError: true,
+		},
+		{
+			Name:       "curl error",
+			Args:       []string{requestprocessorName},
+			ExecHelper: "RequestProcessorInvokeError",
+			GivenObjects: []runtime.Object{
+				requestProcessor,
+				ingressService,
+			},
+			ShouldError: true,
+			Verify: func(t *testing.T, output string, err error) {
+				for _, expected := range []string{
+					"curl localhost -H 'Host: test-requestprocessor.example.com'\n",
+					"exit status 255\n",
+				} {
+					if !strings.Contains(output, expected) {
+						t.Errorf("expected command output to contain %q, actually %q", expected, output)
+					}
+				}
+			},
+		},
+	}
+
+	table.Run(t, commands.NewRequestProcessorInvokeCommand)
+}
+
+func TestHelperProcess_RequestProcessorInvoke(t *testing.T) {
+	if os.Getenv("GO_WANT_HELPER_PROCESS") != "1" {
+		return
+	}
+	// TODO assert arguments
```

Comment:
Is this TODO to be done in this PR?

Reply:
The TODO can be removed. Because the args are logged on the next line, they can be asserted in the mainline test. For example https://github.com/projectriff/riff/pull/1269/files#diff-97f91502cd33a30cf14d894726f823c5R369